### PR TITLE
[4.8] Avoid using function each()

### PR DESCRIPTION
**Pull request by mlocati**
Backport of 48930fc4782adf95037ee0ed5440b7366a048c9e by @eclipxe13 to the 4.8 series

### CODE REVIEW

**Comment by SpacePossum** [comment]
cherry-pick of that commit didn't work?

**Comment by eclipxe13** [comment]
I think other commit is missing, the full PR is about 4 commits, 2 of tests and 2 of code.
https://github.com/sebastianbergmann/phpunit/pull/2536/commits

**Comment by SpacePossum** [comment]
maybe try picking the 4? having the tests would be nice in this branch as well I guess :)

**Comment by mlocati** [comment]
@eclipxe13 You are right!

@SpacePossum I manually rewrote [the 4th commit](https://github.com/sebastianbergmann/phpunit/pull/2536/commits/4c37622a51010e5bf5a418d8131c1fd2732d39be), cherry picked the two commits for tests, and fixed the tests for older PHP and PHPUnit

File: tests/Util/GetoptTest.php [hunk 1]
```diff
@@ -59,4 +59,184 @@ public function testItIncludeTheShortOptionsAfterTheArgument()
...
+            $exception = $x;
+        }
+        $this->assertNotNull($exception);
+        $this->assertInstanceOf('PHPUnit_Framework_Exception', $exception);
+        $this->assertSame("option --foo doesn't allow an argument", $exception->getMessage());
+    }
+
+    public function testItHandlesLongParametesWithValues()
+    {
+        $command = 'command parameter-0 --exec parameter-1 --conf config.xml --optn parameter-2 --optn=content-of-o parameter-n';
```
[the remark this answers is not in the document]
Comment (mlocati) [reply tests/Util/GetoptTest.php]: Fixed, thanks bot :wink:

**Comment by mlocati** [comment]
> cherry-pick of that commit didn't work?

I had a merge conflict, so I preferred to rewrite it by hand

**Comment by Remo** [comment]
Yes please, that would fix the build for the legacy version of concrete5 https://github.com/concrete5/concrete5-legacy/

**Comment by sebastianbergmann** [comment]
PHPUnit 4.8 is no longer supported. Please open a new ticket in case this still applies to a supported version of PHPUnit.

**Comment by mlocati** [comment]
@sebastianbergmann We need to test a software that supports PHP from version 5.3 to 7.2 (dropping end-of-life PHP versions is not an option for us). Do you have any suggestion about how could we do that?

**Comment by hboomsma** [comment]
You could always fork your preferred version of PHPUnit and maintain it yourself since no new commits will be put in the upstream one.

**Comment by mlocati** [comment]
@hboomsma thanks for the feedback. I already know we can fork phpunit, but I was wondering if @sebastianbergmann had a better approach, since I think we are not the only one with this need...

**Pull request closed without merging**